api/config.py

```diff
@@ -262,11 +280,14 @@ def get_config():
     return __config
 
 def get_public_config():
+    auth = copy.deepcopy(__config.get('auth'))
+    for values in auth.itervalues():
+        values.pop('client_secret', None)
```

Comment:
It's a blacklist inside of a white list, L I S T C E P T I O N.

The white list for fields inside an auth config would be a lot longer than the blacklist, and is not standard for each type of config. I figured this was more clear, but I see your point.